### PR TITLE
Removed the while loop in triggerPruneOnThresholdBreach

### PR DESCRIPTION
**Pull request by NehaNaithani**

### CODE REVIEW

**Review by baghbidi — APPROVED** [review]

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java [hunk 1]
```diff
@@ -72,12 +74,19 @@ public void start() throws Exception {
...
             long availableDiskPercentage = StatusReporter.getResourceConsumptionManagerStatus().getAvailableDisk() * 100 /
                     StatusReporter.getResourceConsumptionManagerStatus().getTotalDiskSpace();
-            if (Configuration.getAvailableDiskThreshold() >= availableDiskPercentage){
-                LoggingService.logDebug(MODULE_NAME, "Docker Prune when available disk is equal to or less than threshold");
-                docker.dockerPrune();
+            if (availableDiskPercentage < Configuration.getAvailableDiskThreshold()) {
+                try {
+                    LoggingService.logDebug(MODULE_NAME, "Docker Prune when available disk is less than threshold");
+                    isPruning = true;
+                    PruneResponse response = docker.dockerPrune();
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java]: `response` is not being used, right?

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java [hunk 2]
```diff
@@ -72,12 +74,19 @@ public void start() throws Exception {
...
+                    isPruning = false;
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java]: set `isPruning` to `false` in `finally` block. Then there is no need to have it in both `try` and `catch`

**Review by baghbidi — CHANGES_REQUESTED** [review]

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java (hunk 1 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java]: yeah it's not being used. Will remove it.

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java (hunk 2 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/pruning/DockerPruningManager.java]: I always forget the finally block. Will update it thanks

**Review by baghbidi — APPROVED** [review]

**Pull request merged**